[22.01] Wrap reports app in static, fix tool shed entry point

mvdbeek opened the pull request:
Fixes https://github.com/galaxyproject/galaxy/issues/13400.

We should document `GUNICORN_CMD_ARGS` too or add some other way to set the ports for the reports and tool_shed app, but this is the minimal fix.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:

Start tool_shed (`./run_tool_shed.sh` ) and reports (`./run_reports.sh`) and make sure you can reach both apps on their default ports (9009 and 9001 respectively).

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

mvdbeek committed:
Wrap reports and tool_shed app in static

That broke in https://github.com/galaxyproject/galaxy/pull/13103,
but we only noticed it when we started the apps with gunicorn
instead of uwsgi, since uwsgi has its own static mapping.

mvdbeek committed:
Fix tool_shed entrypoint and default port

jdavcs commented on scripts/common_startup_functions.sh:
```diff
@@ -187,9 +187,14 @@ find_server() {
+        else
+            export GUNICORN_CMD_ARGS="${GUNICORN_CMD_ARGS:-\"--bind=localhost:8080\"}"
+            server_args="'galaxy.webapps.${server_app}.fast_factory:factory()' --pythonpath lib -k ${gunicorn_worker:-$default_gunicorn_worker} $gunicorn_args"
+        fi
```
I think we can simplify this by setting the port in one place. We set the port for reports on line 166; if port selection is not dependent on which server will run, let's just move port selection out of this if-statement. `server_args`, on the other hand, is server-dependent, so it should stay here; and aren't we missing setting `server_args` for reports in the same way we are setting it for tool_shed?

mvdbeek replied:
I've moved the port setting. Reports is in galaxy.webapps, so that is correct.

mvdbeek committed:
Move up GUNICORN_CMD_ARGS

mvdbeek committed:
Allow starting shed and reports apps from sample configs

davelopez reviewed (approved):
Thank you @mvdbeek It works nicely out of the box! :)

Not really a concern, and nothing to be addressed here, but it will be nice to show the listening address and port like in `Listening at: http://127.0.0.1:9001` right after `[INFO] Application startup complete` just for lazy click convenience :wink: 

```
[2022-02-23 11:58:47 +0100] [9435] [INFO] Starting gunicorn 20.1.0
[2022-02-23 11:58:47 +0100] [9435] [INFO] Listening at: http://127.0.0.1:9001 (9435)
[2022-02-23 11:58:47 +0100] [9435] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2022-02-23 11:58:47 +0100] [9436] [INFO] Booting worker with pid: 9436
...
(potentially lots of log lines here)
...
[2022-02-23 11:58:49 +0100] [9436] [INFO] Started server process [9436]
[2022-02-23 11:58:49 +0100] [9436] [INFO] Waiting for application startup.
[2022-02-23 11:58:49 +0100] [9436] [INFO] Application startup complete.
```

nsoranzo commented on scripts/common_startup_functions.sh:
```diff
@@ -163,8 +163,10 @@ find_server() {
             ;;
         reports)
             # TODO: don't override this var, and is this really the only way to configure the port?
```
```suggestion
            # TODO: is this really the only way to configure the port?
```

mvdbeek committed:
Tweak comment

And no, we could read out the port from the uwsgi section, or decide to move it elsewhere.

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo commented on scripts/common_startup_functions.sh:
```diff
@@ -162,9 +162,11 @@ find_server() {
-            # TODO: don't override this var, and is this really the only way to configure the port?
-            GUNICORN_CMD_ARGS="--bind=localhost:9001"
+            # TODO: is this really the only way to configure the port?
+            GUNICORN_CMD_ARGS=${GUNICORN_CMD_ARGS:-\"--bind=localhost:9001\"}
```
```suggestion
            GUNICORN_CMD_ARGS=${GUNICORN_CMD_ARGS:-"--bind=localhost:9001"}
```

nsoranzo commented on scripts/common_startup_functions.sh:
```diff
@@ -162,9 +162,11 @@ find_server() {
+            GUNICORN_CMD_ARGS=${GUNICORN_CMD_ARGS:-\"--bind=localhost:9001\"}
             ;;
+        tool_shed)
+            GUNICORN_CMD_ARGS=${GUNICORN_CMD_ARGS:-\"--bind=localhost:9009\"}
```
```suggestion
            GUNICORN_CMD_ARGS=${GUNICORN_CMD_ARGS:-"--bind=localhost:9009"}
```

mvdbeek committed:
Don't escape quotes

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>
